alpha/veneer/semver/semver.go

```diff
@@ -434,3 +445,41 @@ func MermaidChannelWriter(cfg declcfg.DeclarativeConfig, out io.Writer) error {
 	}
 	return nil
 }
+
+func withoutBuildMetadataConflict(versions *map[string]semver.Version) error {
+	errs := []error{}
+
+	// using the stringified semver because the semver package generates deterministic representations,
+	// and because the semver.Version contains slice fields which make it unsuitable as a map key
+	//      stringified-semver.Version ==> incidence count
+	seen := make(map[string]int)
```

Comment:
It seems like we could have a map of string -> struct{} or string -> bool k/v pairs here instead. Are we consuming the number of seen semver strings anywhere, or are we just checking for whether we've seen and processing this semver string before?

Reply:
We're not consuming the incidence anywhere, but this is me #futureproofing.  I figure if we get to a place where we want to report the collision count as well, we're fortified.